src/timer.rs

```diff
@@ -105,6 +107,36 @@ macro_rules! hal {
                         Event::Update => self.tim.dier.write(|w| w.uie().clear_bit()),
                     }
                 }
+
+                /// Return the bus clock frequency in hertz.
+                fn get_bus_clock(&self) -> Hertz {
+                    if TypeId::of::<$apbX>() == TypeId::of::<APB1>() {
+                            Hertz(self.clocks.pclk1().0 * self.get_bus_frequency_multiplier())
+                    } else if TypeId::of::<$apbX>() == TypeId::of::<APB2>() {
+                        Hertz(self.clocks.pclk2().0 * self.get_bus_frequency_multiplier())
+                    } else {
+                        unreachable!()
+                    }
+                }
+
+                /// Return the bus frequency multiplier.
+                fn get_bus_frequency_multiplier(&self) -> u32 {
+                    if TypeId::of::<$apbX>() == TypeId::of::<APB1>() {
```

Comment:
Just curious: Any reason not to use `match`?

Reply:
Not at all...

Reply:
Never mind, just tried it, matching does not really work easily.